src/librustdoc/config.rs

```diff
@@ -496,6 +498,21 @@ impl Options {
         let enable_per_target_ignores = matches.opt_present("enable-per-target-ignores");
         let document_private = matches.opt_present("document-private-items");
         let document_hidden = matches.opt_present("document-hidden-items");
+        let source_code_external_url = matches.opt_str("source-code-external-url").map(|mut h| {
+            if !h.ends_with('/') {
+                h.push('/');
+            }
+            h.replace("\\", "/")
+        });
+        if let Some(ref source_code_external_url) = source_code_external_url {
+            if source_code_external_url.starts_with("/") {
+                diag.struct_err(
+                    "option `--source-code-external-url` argument cannot be an absolute local path",
```

Comment:
Why this limitation?

Reply:
Actually, after thinking about it, it doesn't make much sense... I'll remove it.